csaf_2.0/json_schema/csaf_json_schema.json

```
@@ -154,13 +154,127 @@
+          "type": "object",
+          "minProperties": 1,
+          "properties": { 
+            "cpe": {
+              "$comment": "TODO: This should have a full regular expression to enforce CPE syntax.",
+              "title": "Common Platform Enumeration representation",
+              "description": "The Common Platform Enumeration (CPE) attribute refers to a method for naming platforms external to this specification.",
+              "type": "string",
+              "pattern": "^(cpe:2\\.3:[aho\\*\\-](:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){5}(:(([a-zA-Z]{2,3}(-([a-zA-Z]{2}|[0-9]{3}))?)|[\\*\\-]))(:(((\\?*|\\*?)([a-zA-Z0-9\\-\\._]|(\\\\[\\\\\\*\\?!\"#\\$%&'\\(\\)\\+,/:;<=>@\\[\\]\\^`\\{\\|\\}~]))+(\\?*|\\*?))|[\\*\\-])){4})|([c][pP][eE]:/[AHOaho]?(:[A-Za-z0-9\\._\\-~%]*){0,6})$",
+              "minLength": 5
+            },
+            "hashes": {
+              "$comment": "TODO: Array of objects (algorithm, value, optional name)",
+              "title": "List of hashes",
+              "description": "Contains a list of cryptographic hashes usable to identify files.",
+              "type": "array",
+              "minItems": 1,
+              "items": {
+                "title": "Cryptographic hash",
+                "description": "Contains all information to identify a file based on its cryptographic hash value.",
+                "type": "object",
+                "required": [
+                  "algorithm",
+                  "file",
+                  "value"
+                ],
+                "properties": {
+                  "algorithm": {
+                    "title": "Algorithm of the cryptographic hash",
+                    "description": "Contains the name of the cryptographic hash algorithm used to calculate the value.",
+                    "type": "string",
+                    "enum": [
+                      "SHA-256",
+                      "SHA-384",
+                      "SHA-512",
+                      "SHA-3",
+                      "BLAKE3"
+                    ]
+                  },
+                  "file": {
+                    "title": "Filename",
+                    "description": "Contains the filename of the file which is identified by the hash value.",
+                    "type": "string",
+                    "minLength": 1,
+                    "examples": [
+                      "WINWORD.EXE",
+                      "msotadddin.dll",
+                      "sudoers.so"
+                    ]
+                  },
+                  "value": {
+                    "title": "Value of the cryptographic hash",
+                    "description": "Contains the cryptographic hash value.",
+                    "type": "string",
+                    "minLength": 64,
+                    "pattern": "^([0-9a-fA-F]){64,}$",
```

The capturing group is not really necessary here in my opinion:

```suggestion
                    "pattern": "^[0-9a-fA-F]{64,}$",
```